### PR TITLE
TMA-1033: show reason of filter composition failure

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 1]
```diff
@@ -210,47 +210,53 @@ def self.get_small_labels(labels_cache)
...
+                       nil
+                     end
+                   elsif no_values
+                     # create a filter that is always true to ensure the user can see all data
                      'TRUE'
                    elsif filter[:over] && filter[:to]
                      over = attr_cache[filter[:over]]
                      to = attr_cache[filter[:to]]
-                     "([#{label.attribute_uri}] IN (#{element_uris.compact.sort.map { |e| '[' + e + ']' }.join(', ')})) OVER [#{over && over.uri}] TO [#{to && to.uri}]"
+                     "([#{label.attribute_uri}] IN (#{element_uris.sort.map { |e| '[' + e + ']' }.join(', ')})) OVER [#{over && over.uri}] TO [#{to && to.uri}]"
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Calls ''[' + e + ']'' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Calls ''[' + e' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Calls 'element_uris.sort' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Calls 'element_uris.sort.map { |e| '[' + e + ']' }' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Calls 'element_uris.sort.map { |e| '[' + e + ']' }.join(', ')' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Calls 'element_uris.sort.map' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Calls 'label.attribute_uri' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Has the variable name 'e' (UncommunicativeVariableName) (reek)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 2]
```diff
@@ -210,47 +210,53 @@ def self.get_small_labels(labels_cache)
...
-          errors << {
-            type: :error,
-            label: label.title,
-            value: v
-          }
-          nil
+      errors = []
+
+      element_uris_by_values = Hash[values.map do |v|
+        if lookups_cache.key?(label.uri)
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Calls 'label.uri' 2 times (DuplicateMethodCall) (reek)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 3]
```diff
@@ -210,47 +210,53 @@ def self.get_small_labels(labels_cache)
...
-                   elsif element_uris.compact.empty? && options[:restrict_if_missing_all_values] && options[:type] == :variable
-                     nil
-                   elsif element_uris.compact.empty?
+      errors += missing_value_errors unless options[:ignore_missing_values]
+
+      element_uris = element_uris_by_values.values.compact
+      # happens when data is not yet loaded in the project
+      no_values = element_uris.empty?
+
+      expression = if no_values && options[:restrict_if_missing_all_values] && options[:type] == :muf
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Calls 'options[:type]' 2 times (DuplicateMethodCall) (reek)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 4]
```diff
@@ -210,47 +210,53 @@ def self.get_small_labels(labels_cache)
...
+
+      element_uris_by_values = Hash[values.map do |v|
+        if lookups_cache.key?(label.uri)
+          [v, lookups_cache[label.uri][v]]
+        else
+          [v, label.find_value_uri(v)]
         end
+      end]
+
+      missing_value_errors = element_uris_by_values.select { |_, v| v.nil? }.map do |k, _|
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Performs a nil-check (NilCheck) (reek)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Has the variable name 'k' (UncommunicativeVariableName) (reek)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 5]
```diff
@@ -210,47 +210,53 @@ def self.get_small_labels(labels_cache)
...
-        rescue
-          errors << {
-            type: :error,
-            label: label.title,
-            value: v
-          }
-          nil
+      errors = []
+
+      element_uris_by_values = Hash[values.map do |v|
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Has the variable name 'v' (UncommunicativeVariableName) (reek)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/558/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1390/ : SUCCESS in 3m 21s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1660/ : SUCCESS in 2m 34s

**Comment by kubamahnert** [comment]
ok to test

**Comment by kubamahnert** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/559/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1391/ : SUCCESS in 53s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1661/ : SUCCESS in 1m 06s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/560/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1508/ : FAILURE in 23m 27s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/79/ : SUCCESS in 27m 52s

**Comment by kubamahnert** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/581/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1520/ : FAILURE in 26m 14s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/89/ : SUCCESS in 34m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/583/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1405/ : SUCCESS in 2m 28s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1675/ : SUCCESS in 1m 08s

**Comment by kubamahnert** [comment]
extended test

**Pull request closed without merging**